poetry/packages/locker.py

```diff
@@ -271,12 +279,13 @@ def __walk_dependency_level(
                 ].marker.intersect(requirement.marker)
 
         return cls.__walk_dependency_level(
-            dependencies=next_level_dependencies,
+            dependencies=list(next_level_dependencies.values()),
```

Comment:
```suggestion
            dependencies=next_level_dependencies.values(),
```

Not required, but might be nice to have since we iterate over this anway.